plugins/updater/src/updater.rs

```diff
@@ -271,11 +280,15 @@ impl Updater {
                 .replace("{{arch}}", self.arch)
                 .parse()?;
 
-            let mut request = Client::new().get(url).headers(headers.clone());
+            let mut request = ClientBuilder::new();
             if let Some(timeout) = self.timeout {
                 request = request.timeout(timeout);
             }
-            let response = request.send().await;
+            if let Some(ref proxy) = self.proxy {
+                let proxy = reqwest::Proxy::all(proxy.as_str())?;
```

Comment:
I believe `Proxy::all` can take `Url` directly so we don't need to do `.as_str()`, no?

Reply:
It needs to be `reqwest::Proxy::all(proxy.clone())` or `reqwest::Proxy::all(proxy.as_str())`, I took the later to avoid copy which I think performance may be better

Reply:
It needs to be either reqwest::Proxy::all(proxy.clone()) or reqwest::Proxy::all(proxy.as_str()), I took the later to avoid copy which I think performance may be better. Which do you prefer?

Reply:
My bad, I guess you need to do `.clone()` because of `ref` and we can't move it out because it will be used again later.

> which I think performance should be better
it shouldn't matter since `Proxy::all` will take the `&str` and call `Url::parse` on it so let's just stick with your choice.